src/python/dxpy/utils/app_unbuilder.py

```diff
@@ -144,7 +161,7 @@ def dump_executable(executable, destination_directory, omit_resources=False, des
 
         # Write dxapp.json, Readme.md, and Readme.developer.md
         with open("dxapp.json", "w") as f:
-            f.write(json.dumps(dxapp_json, sort_keys=False, indent=2, separators=(',', ': ')))
+            f.write(flatten_patterns(json.dumps(dxapp_json, sort_keys=True, indent=2, separators=(',', ': '))))
```

Comment:
The idea here is to print the items in the order that the user is likely to want to read them. dx-app-wizard implements something similar by using an OrderedDict and setting fields in the following order:

```
name
title
summary
dxapi
version
...
```

Have a look at that file: https://github.com/dnanexus/dx-toolkit/blob/master/src/python/scripts/dx-app-wizard to get the keys in an order you can emulate.